plugins/woocommerce-admin/client/customize-store/assembler-hub/sidebar/pattern-screen/sidebar-pattern-screen.tsx

```diff
@@ -0,0 +1,146 @@
+/**
+ * External dependencies
+ */
+import { useCallback, useMemo, useRef, useState } from '@wordpress/element';
+import { useSelect, useDispatch, select } from '@wordpress/data';
+import { cloneBlock } from '@wordpress/blocks';
+import { close } from '@wordpress/icons';
+import { __ } from '@wordpress/i18n';
+import { getNewPath, navigateTo } from '@woocommerce/navigation';
+import { capitalize } from 'lodash';
+import { Button, Spinner } from '@wordpress/components';
+import {
+	unlock,
+	// @ts-expect-error No types for this exist yet.
+} from '@wordpress/edit-site/build-module/lock-unlock';
+
+// eslint-disable-next-line @woocommerce/dependency-group
+import {
+	store as coreStore,
+	// @ts-expect-error No types for this exist yet.
+} from '@wordpress/core-data';
+// eslint-disable-next-line @woocommerce/dependency-group
+import {
+	__experimentalBlockPatternsList as BlockPatternList,
+	store as blockEditorStore,
+	// @ts-expect-error No types for this exist yet.
+} from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import { usePatternsByCategory } from '../../hooks/use-patterns';
+import './style.scss';
+import { useEditorBlocks } from '../../hooks/use-editor-blocks';
+import { PATTERN_CATEGORIES } from './categories';
+
+export const SidebarPatternScreen = ( { category }: { category: string } ) => {
```

Comment:
Do you mind clarifying why we aren't using Gutenberg's InserterSidebar component here? https://github.com/WordPress/gutenberg/blob/550cb6b67b544c72d124e8d6741e0d5577da9256/packages/editor/src/components/inserter-sidebar/index.js#L20C25-L20C40